pkg/controller/nodedaemon/reconcile.go

```diff
@@ -67,5 +83,51 @@ func (r *DaemonReconciler) Reconcile(request reconcile.Request) (reconcile.Resul
 		r.reqLogger.Info("daemonset changed", "daemonset.Name", ds.GetName(), "op.Result", opResult)
 	}
 
-	return reconcile.Result{}, nil
+	return reconcile.Result{}, err
+}
+
+// do a one-time delete of the old static-provisioner daemonset
+func (r *DaemonReconciler) cleanupOldDaemonsets(namespace string) error {
+	if r.deletedStaticProvisioner {
```

Comment:
Is it possible that this function can be called again and again while we wait for daemonset to be deleted? I think that - we should kind of return from here or do something else - rather than spamming the API server.

Reply:
The delete function should only be called once. Once the daemonset is gone from the api server, it simply queries for pods that the daemonset creates with reasonable intervals (even a backoff).